src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -1845,108 +1788,201 @@ impl<'a> TypeChecker<'a> {
-                        )));
+                let idx = match path_lit {
+                    Literal::Integer(idx) => {
+                        if idx == 0 {
+                            return Err(ErrorCode::SemanticError(
+                                "tuple index is starting from 1, but 0 is found".to_string(),
+                            ));
+                        }
+                        if idx as usize > fields_type.len() {
+                            return Err(ErrorCode::SemanticError(format!(
+                                "tuple index {} is out of bounds for length {}",
+                                idx,
+                                fields_type.len()
+                            )));
+                        }
+                        table_data_type = fields_type.get(idx as usize - 1).unwrap().clone();
+                        idx as usize
                     }
-                    let inner_name = inner_names.get(idx as usize - 1).unwrap();
-                    let inner_type = inner_types.get(idx as usize - 1).unwrap();
-                    names.push(inner_name.clone());
-                    data_types.push(inner_type.clone());
+                    Literal::String(name) => match fields_name.iter().position(|k| k == &name) {
+                        Some(idx) => {
+                            table_data_type = fields_type.get(idx).unwrap().clone();
+                            idx + 1
+                        }
+                        None => {
+                            return Err(ErrorCode::SemanticError(format!(
+                                "tuple name `{}` does not exist, available names are: {:?}",
+                                name, &fields_name
+                            )));
+                        }
+                    },
+                    _ => unreachable!(),
+                };
+                scalar = FunctionCall {
+                    params: vec![idx],
+                    arguments: vec![scalar.clone()],
+                    func_name: "get".to_string(),
+                    return_type: Box::new(DataType::from(&table_data_type)),
                 }
-                Literal::String(name) => match inner_names.iter().position(|k| k == &name) {
-                    Some(idx) => {
-                        let inner_name = inner_names.get(idx).unwrap();
-                        let inner_type = inner_types.get(idx).unwrap();
+                .into();
+                continue;
+            }
+            let box (path_value, path_data_type) = self.resolve_literal(&path_lit, None)?;
+            let path_scalar: Scalar = ConstantExpr {
+                value: path_value,
+                data_type: Box::new(path_data_type.clone()),
+            }
+            .into();
+            if let TableDataType::Array(inner_type) = table_data_type {
+                table_data_type = *inner_type;
+            }
+            table_data_type = TableDataType::wrap_nullable(&table_data_type);
```

Comment:
Why wrap_nullable？

Reply:
`get` function for `variant` and `array` will always return a nullable value, because the index may out of range.